File: solutions/SquareScoresDiv2/solve/javascript/SquareScoresDiv2Solver.js
#! /usr/bin/env node

var fs = require("fs");
var path = require("path");
var tc = null;

function init() {
    var gettcHome = process.env.GETTC_HOME || path.join(process.env.HOME, ".gettc");
    var includeDir = path.join(gettcHome, "include", "javascript");
    tc = require(path.join(includeDir, "topcoder"));
}

function main() { try {
    var input = fs.readFileSync(process.argv[2], { encoding: "ascii" });
    var reader = new tc.Reader(input);
    var s = reader.next("String");

    var SquareScoresDiv2 = require("./SquareScoresDiv2");
    var result = SquareScoresDiv2.getscore(s);
    
    var writer = new tc.Writer();
    writer.next(result, "int");
    fs.writeFileSync(process.argv[3], writer.toString(), { encoding: "ascii" });
} catch (err) {
    console.log(err.toString());
    console.trace();
}}

init();
main();

File: SquareScoresDiv2/solve/ruby/SquareScoresDiv2Solver.rb
#! /usr/bin/env ruby

require_relative "SquareScoresDiv2"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    s = reader.next(TString)

    result = SquareScoresDiv2.new().getscore(s)
    IO.write(ARGV[1], Writer.new().next(result, TInt).to_s())
end
init()
main()
